tests/ui/functions.rs

```diff
@@ -1,5 +1,5 @@
 #![warn(clippy::all)]
-#![allow(dead_code)]
+#![allow(dead_code, improper_ctypes)]
```

Comment:
This was already addressed in #4781. Same with `needless_pass_by_value.rs` test.